bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```diff
@@ -0,0 +1,403 @@
+                }
+            } else {
+                Assignment assignment = (Assignment) abstractAssignment;
+                String value = behaviorConverter.termToString(assignment.getTerm());
+
+                for (Label label : assignment.getOutputLabels()) {
+                    try {
+                        builder.append("set ").append(((LabelType) label.eContainer()).getEntityName()).append(".").append(label.getEntityName())
+                                .append(" = ").append(value).append("\n");
+                    } catch (IllegalArgumentException ex) {
+                        logger.error("Illegal behavior argument");
+                    }
+                }
+            }
+        }
+        return builder.toString().trim();
+    }
+
+    private void parseBehavior(Node node, Pin outpin, String lines, DataFlowDiagram dfd, DataDictionary dd) {
+        String[] behaviorStrings = lines.split("\n");
+        var behavior = node.getBehaviour();
+        for (String behaviorString : behaviorStrings) {
+            if (behaviorString.contains("forward ")) {
+                String packet = behaviorString.split(" ")[1];
+
+                Pin inpin = dfd.getFlows().stream().filter(flow -> flow.getDestinationNode() == node)
+                        .filter(flow -> flow.getEntityName().equals(packet)).map(Flow::getDestinationPin).findAny().orElse(null);
+
+                var assignment = ddFactory.createForwardingAssignment();
+                assignment.setOutputPin(outpin);
+                assignment.getInputPins().add(inpin);
+                behavior.getAssignment().add(assignment);
+            } else if (behaviorString.contains("set ")) {
+                String regex = "\\bset\\s+(\\S+)\\s+=\\s+(.+)\\b";
+
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(behaviorString);
+                if (matcher.find() && matcher.groupCount() == 2) {
+                    String variable = matcher.group(1);
+                    String typeName = variable.split("\\.")[0];
+                    String valueName = variable.split("\\.")[1];
+
+                    Label value = dd.getLabelTypes().stream().filter(labelType -> labelType.getEntityName().equals(typeName))
+                            .flatMap(labelType -> labelType.getLabel().stream()).filter(label -> label.getEntityName().equals(valueName)).findAny()
+                            .orElse(null);
+
+                    Assignment assignment = ddFactory.createAssignment();
+
+                    assignment.getInputPins().addAll(behavior.getInPin());
+                    assignment.setOutputPin(outpin);
+                    assignment.getOutputLabels().add(value);
+
+                    behavior.getAssignment().add(assignment);
+
+                    Term term = behaviorConverter.stringToTerm(matcher.group(2));
+
+                    assignment.setTerm(term);
+                }
+            }
```

Comment:
Replace if with guard expression:
Check if matcher.find() or matcher.groupCount() == 2 is false and return an error and stop parsing